plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/UriTemplateServerProxyTransformer.java

```diff
@@ -64,21 +70,46 @@ public ServerImpl transform(ServerImpl server) {
         if (serverPath.startsWith("/")) {
             serverPath = serverPath.substring(1);
         }
-
+        final String externalAddress;
+        final String externalServerHost;
+        final String internalServerHost;
+        if (!Strings.isNullOrEmpty(externalAddressProperty) && !Strings.isNullOrEmpty(codenvyHost)) {
```

Comment:
yes it is always set in dockerized codenvy.
I will try if I only rewrite proxytransformer without changing CodenvyDockerServerEvaluationStrategy for handling multiple nodes of swarm (on mac/windows)